File: src/main/java/cn/dev/Learn01.java
package cn.dev;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaSparkContext;
import org.datasyslab.geospark.enums.FileDataSplitter;
import org.datasyslab.geospark.spatialRDD.PointRDD;

public class Learn01 {
    public static void main(String[] args) {
        SparkConf conf = new SparkConf();
        conf.setAppName("GeoSpark01");
        conf.setMaster("local[*]");
        conf.set("spark.serializer", "org.apache.spark.serializer.KryoSerializer");
        conf.set("spark.kryo.registrator", "org.datasyslab.geospark.serde.GeoSparkKryoRegistrator");
        JavaSparkContext sc = new JavaSparkContext(conf);

        // Get SpatialRDD
        String pointRDDInputLocation = Learn01.class.getResource("/checkin.csv").toString();
        Integer pointRDDOffset = 0; // 地理位置(经纬度)从第0列开始
        FileDataSplitter pointRDDSplitter = FileDataSplitter.CSV;
        Boolean carryOtherAttributes = true; // 第二列的属性(酒店名)
        PointRDD rdd = new PointRDD(sc, pointRDDInputLocation, pointRDDOffset, pointRDDSplitter, carryOtherAttributes);

        // 坐标系转换
        String sourceCrsCode = "epsg:4326";
        String targetCrsCode = "epsg:3857";
        rdd.CRSTransform(sourceCrsCode, targetCrsCode);

        sc.stop();
    }
}

File: src/main/scala/cn/dev/scala/Learn01.scala
package cn.dev.scala

import org.apache.spark.{SparkConf, SparkContext}
import org.datasyslab.geospark.enums.FileDataSplitter
import org.datasyslab.geospark.spatialRDD.PointRDD

object Learn01 extends App{
  val conf = new SparkConf
  conf.setAppName("GeoSpark01")
  conf.setMaster("local[*]")
  conf.set("spark.serializer", "org.apache.spark.serializer.KryoSerializer")
  conf.set("spark.kryo.registrator", "org.datasyslab.geospark.serde.GeoSparkKryoRegistrator")
  val sc = new SparkContext(conf)

  val pointRDDInputLocationn = this.getClass.getResource("/checkin.csv").toString
  val pointRDDOffset = 0
  val pointRDDSplitter = FileDataSplitter.CSV
  val carryOtherAttributes = true

  //  implicit def fromSparkContext(sc: SparkContext): JavaSparkContext = new JavaSparkContext(sc)
  //  Scala提供了从SparkContext到JavaSparkContext的隐士转化
  val rdd = new PointRDD(sc, pointRDDInputLocationn, pointRDDOffset, pointRDDSplitter, carryOtherAttributes)

  // 坐标转换
  rdd.CRSTransform("EPSG:4326", "EPSG:4457")

  // rdd.rawSpatialRDD中是JavaRDD,若用Scala,需要RDD,否则会出现很多参数匹配错误
  rdd.rawSpatialRDD.rdd.foreach(println)
}
